### PR TITLE
REF use cph to extract the pinnings package and account for .conda

### LINKED ISSUE #1692: Handle `conda-forge-pinning` in `.conda` format (maybe with `.tar.bz2 fallback handling)
Currently conda-smithy expects `conda-forge-pinning` to be in `.tar.bz2` format (please see below):

https://github.com/conda-forge/conda-smithy/blob/6de36e002b2d7982d77d4baa2a30e7001a4152ef/conda_smithy/configure_feedstock.py#L2019

Would be good to update this to support `.conda` or maybe better both formats. This way `conda-forge-pinning` can be moved to `.conda` as well

### PR DESCRIPTION
**Pull request by beckermr**
Checklist
* [x] Added a ``news`` entry

closes #1692

### CODE REVIEW

**Comment by beckermr** [comment]
I need to test this live on a feedstock to make sure it is still working.

**Comment by beckermr** [comment]
This works fine locally.

**Comment by beckermr** [comment]
@conda-forge/core this is ready for review!

File: conda_smithy.recipe/meta.yaml [hunk 1]
```diff
@@ -22,6 +22,7 @@ requirements:
     - python >=3.6
     - conda >=4.2
     - conda-build >=3.18.3
+    - conda-package-handling
```
Comment (beckermr) [inline conda_smithy.recipe/meta.yaml:25]:
```suggestion
    - conda-package-handling >=1.9.0
```

**Review by chrisburr — APPROVED** [review]

**Pull request merged**